server/routes/auth.ts

```diff
@@ -40,121 +41,63 @@ app.use(session({
 	saveUninitialized: false
 }));
 passport.serializeUser<IUser, string>((user, done) => {
-	done(null, user._id.toString());
+	done(null, user.uuid);
 });
 passport.deserializeUser<IUser, string>((id, done) => {
-	User.findById(id, (err, user) => {
+	User.findOne({ uuid: id }, (err, user) => {
 		done(err, user!);
 	});
 });
 
-let router = express.Router();
-export let authRoutes: express.RequestHandler = (request, response, next) => {
-	// Allows for dynamic dispatch when authentication gets reloaded
-	router(request, response, next);
-};
+export let authRoutes = express.Router();
 
-export async function reloadAuthentication() {
-	router = express.Router();
-	let authenticationMethods: RegistrationStrategy[] = [];
-	let methods = await getSetting<(keyof typeof strategies)[]>("loginMethods");
-	console.info(`Using authentication methods: ${methods.join(", ")}`);
-	for (let methodName of methods) {
-		if (!strategies[methodName]) {
-			console.error(`Authentication method "${methodName}" is not available. Did you add it to the exported list of strategies?`);
-			continue;
-		}
-		let method = new strategies[methodName]();
-		authenticationMethods.push(method);
-		method.use(router);
-	}
-
-	// These routes need to be redefined on every instance of a new router
-	router.post("/confirm", validateAndCacheHostName, postParser, async (request, response) => {
-		let user = request.user as IUserMongoose;
-		let name = request.body.name as string;
-		if (!name || !name.trim()) {
-			request.flash("error", "Invalid name");
-			response.redirect("/login/confirm");
-			return;
-		}
-		if (!request.isAuthenticated() || !user) {
-			request.flash("error", "Must be logged in");
-			response.redirect("/login");
-			return;
-		}
-		user.name = name.trim();
-
-		let email = request.body.email as string | undefined;
-		if (email && email !== user.email) {
-			if (!email.trim()) {
-				request.flash("error", "Invalid email");
-				response.redirect("/login/confirm");
-				return;
-			}
-			if (await User.count({ email }) > 0) {
-				request.flash("error", "That email address is already in use. You may already have an account from another login service.");
-				response.redirect("/login/confirm");
-				return;
-			}
-			user.admin = false;
-			user.verifiedEmail = false;
-			user.email = email;
-			if (config.admins.includes(email)) {
-				user.admin = true;
-				console.info(`Adding new admin: ${email}`);
-			}
-		}
-		user.accountConfirmed = true;
+authRoutes.get("/validatehost/:nonce", (request, response) => {
```

Comment:
What is the purpose of this? What does this achieve that's not handled by the oauth client library

Reply:
This is here because we use the `Host` header when generating links and callback URLs in the auth code. This verifies that the server behind the domain in the `Host` header is actually us and not some malicious other instance. Not sure if it's overengineered because `Host` headers can only be crafted by non-browser HTTP libraries (AJAX requests and JavaScript in browsers cannot override the `Host` header)